django/db/backends/postgresql/operations.py

```diff
@@ -271,5 +272,18 @@ def explain_query_prefix(self, format=None, **options):
             prefix += ' (%s)' % ', '.join('%s %s' % i for i in extra.items())
         return prefix
 
-    def ignore_conflicts_suffix_sql(self, ignore_conflicts=None):
-        return 'ON CONFLICT DO NOTHING' if ignore_conflicts else super().ignore_conflicts_suffix_sql(ignore_conflicts)
+    def on_conflict_suffix_sql(
+        self, opts, fields, on_conflict=None, update_fields=None, unique_fields=None
+    ):
+        if on_conflict == OnConflict.Ignore:
+            return 'ON CONFLICT DO NOTHING'
+
+        if on_conflict == OnConflict.Update:
+            result = 'ON CONFLICT("%s") DO UPDATE SET ' % ('", "'.join(unique_fields))
+            result += ', '.join('%s=excluded.%s' % (field, field) for field in update_fields)
+            return result
```

Comment:
Use `self.quote_name()` to do the quoting of fields:

```suggestion
            return 'ON CONFLICT(%s) DO UPDATE SET %s' % (
                ', '.join(map(self.quote_name, unique_fields or ())),
                ', '.join(f'{field}=excluded.{field}' for field in map(self.quote_name, update_fields or ())),
            )
```